internal/credential/vault/service_library.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// LibraryService coordinates calls to gather information about all credential libraries.
+type LibraryService struct {
+	repo   *Repository
```

Comment:
do we want to define an interface which constrains the dependency the receiver
funcs required?

Reply:
I don't know that it's that useful when the expected type is in the same package. Better to test with a real repository than using a fake/mock, right?